### ISSUE TITLE
How can Dask acknowledge developer contributions by companies?

### ISSUE THREAD

**Issue by dharhas** [dask/community#32]
Moving https://github.com/dask/dask/issues/5831 to here since this is a better fit.

Hi All,

I'd like to start a discussion on appropriate ways and criteria for the Dask community to acknowledge support by commercial companies. I've had a couple of these conversations offline with a couple of people and have some ideas but I wanted to raise it up for a larger discussion.

Currently the foot of the Dask website has a section called 'Supported By' that from my understanding lists organizations and companies that have supported Dask development through some sort of funding/grants etc. This is a very clear cut form of support.

Another form of support is through the funding of developers who directly work on Dask. At Quansight (where I work) we have clients who support a lot of Dask (and other open source) development and are looking for ways to be recognized as a good open source citizen.

My proposal is we develop some loose criteria on how to evaluate a company's contributions and whether they have been a good citizen and then either add their name/logo to the 'supported by' section of the website OR if we think we need to, add a new section to distinguish between direct monetary support and support through developers. I'd love to hear other ideas on how we could recognize these kind of contributions. We have done blog posts in the past but those are kinda ephemeral.

The advantages of this to the Dask community is to hopefully encourage more companies to contribute their resources and for the companies the recognition that they are open source friendly can help them in recruiting.

**Comment by mrocklin** [comment]
Thanks for raising this @dharhas !  I agree that this is an important topic.  

I do think that it is important to differentiate between organizations that donate employee time and funds for general maintenance and organizations that pay for Dask maintainers to fix problems that are critical to them.  Both activities are critical to the health of the project, but the former organizations are set apart in that they're operating slightly more altrusitically (not quite, but closer).  If an organization wanted to join the list of "Supported By" companies then I would want to see them donate something significant to the organization.  This might be either maintainer time to handle issues that are unrelated to their work, or direct money to Dask through NumFOCUS at a level that it allows us to pay for part of a developer (something in the $100k range).  The for-profit corporations currently listed in the Supported By section currently donate around half of an FTE towards general maintenance as well as whatever their employees do in the course of their everyday work.

Eventually I would love to see a "Used by" section that included companies that use Dask regularly and are happy to talk about it.  I would also love to see (but this is by no means manditory) some information there about roughly how they use it and how it solves their problems (to the extent that they're willing to share). This might be text content, or a 5-10 minute video interview.

I think that we can easily put up a decently populated "Used by" section (indeed we already have the "Powered by" section for OSS projects).  But it sounds like what you're asking about @dharhas is something slightly in between, a "Used enough within the organization that they're happy to pay people to solve their problems" section :)

**Comment by TomAugspurger** [comment]
I agree with Matt that the distinction between "General maintenance" and
"Specific maintenance" is important.

I also think that we should be recognizing the institutions that are
providing "specific maintenance".  A subsection
on our "Supported By" makes sense.

**Comment by martindurant** [comment]
Does "specific maintenance" include any company/institution which has paid for the time of one of it's employees to contribute commits to the dask/distributed (or other) codebase? That sounds like a fairly low bar.

**Comment by TomAugspurger** [comment]
In
https://github.com/dask/governance/blob/master/governance.md#institutional-partners,
we state that

> This support must be for the general development and maintenance of the
Project, and benefit a wide variety of users of the Project. This support
must be substantial and sustained, accounting for the employment of the
equivalent of one maintainer, over several months.

So this middle tier would keep the "substantial and sustained" component,
but the contributions would be focused on the specific needs of the
institution, rather than general maintenance.

**Comment by dharhas** [comment]
Yes I think there needs to be a bar to clear like "substantial and sustained". Recognizing folks for a one-off contributions is not the aim of my request. It is exactly this, being able to recognized companies/institutes that have been supporting Dask in a sustained way but maybe not through direct monetary contributions.

I also think @mrocklin idea of having a 'used by' user stories section is a great idea.

**Comment by jrbourbeau** [comment]
Thanks for bringing up this topic @dharhas. 

I agreed that keeping the "substantial and sustained" requirement should solve the situation where an organization says "A developer we employ made a contribution to dask. We would like to be included as a supporter dask.org"

We already have "Institutional" and "Community" partner categories in our governance document. What should specific maintenance partners be called? "Contributing partners"? Other suggestions are welcome

**Comment by mrocklin** [comment]
Would a "Notable Users" be a sufficient category?  This gives some visibility to companies who use the project (and so is probably useful for recruiting and such), but doesn't give them much credit for spending some cycles working on it.  Maybe we have a somewhat vague statement like the following:

# Notable Users

These insitutions' sustained engagement with the project helps to inform and support development.

**Comment by jacobtomlinson** [comment]
That sounds reasonable to me.

Perhaps another approach (in addition) would be to encourage institutions to co-author a blog post on the Dask blog which highlights some Dask feature they are using. This would help in adding more useful content to the blog and give them credit as a notable user. From the reader's perspective it also gives a clear credibility to the post and to Dask without indicating that the institution owns or builds Dask in any way.

**Comment by jcrist** [comment]
I really like Jacob's blog post idea. Many institutions won't be willing/able to share how they use dask though, so we should probably provide another option. I think the verbage Matt used above seems good to me in that respect.

Where would we post information on how to get added to this list? Perhaps along side it we could encourage also submitting a blogpost or adding content to stories.dask.org?

**Comment by jacobtomlinson** [comment]
> Many institutions won't be willing/able to share how they use dask though, so we should probably provide another option.

I did wonder this, perhaps a generic overview of a feature which is important to them would tick that box?

**Comment by martindurant** [comment]
I think there is still a case for recognising institutions that don't just use dask, but contribute back code to our repos - however self-serving those particular changes may be. Other institutions indeed might just be users and support in the sense that they are willing to say "dask is great" publicly.

**Comment by jacobtomlinson** [comment]
Agreed, there seems to be four categories of folks we are discussing here:

- Folks who use Dask
- Folks who fixed a bug which was blocking their work once
- Folks who contribute fixes/enhancements regularly to improve their own experience
- Folks who contribute fixes/enhancements/maintenance/direction for the wider community and sustainability of the project

**Comment by martindurant** [comment]
^ yes, I like that categorisation, except that usually "folks" is a company who allows their employee the freedom to do those things.

**Comment by dharhas** [comment]
I think there is some overlap in the last two. Most institutes are going to focus on parts of the stack are most relevant to their own pain points. The question is are they solely focused on their particular issue or helping with the general improvement of that part of the stack as well.

But in general I think this is a decent segmentation of the group.

**Comment by jacobtomlinson** [comment]
@dharhas I agree that there is an overlap but I think the important distinction is about how invested they are in the health and longevity of the project.
